packages/backend/src/scheduler/SchedulerTask.ts

```diff
@@ -1452,7 +1454,22 @@ export const handleScheduledDelivery = async (
                     savedChartUuid,
                 );
 
-                if (isPositiveThesholdAlert(thresholds, rows)) {
+                if (isPositiveThresholdAlert(thresholds, rows)) {
+                    // If the delivery frequency is once, we disable the scheduler.
+                    // It will get sent once this time.
+                    if (
+                        notificationFrequency === NotificationFrequency.ONCE &&
```

Comment:
Nice

Reply:
this means some scheduler events can happen even when `scheduler` is disabled, like send email/slack notifications. We do the disabled check here https://github.com/lightdash/lightdash/pull/9089/files#diff-76827588cb3d2be92b5d1fec14b04f04caf22bbcd510e49a98517da3192d3fcbR1416 so it should not cause any issues. Just something to keep in mind.

Reply:
Good point. I agree I *think* it should be safe. I could make this just set a local variable like disableAfterSending and disable it at the end of this try block. Would that be better or more confusing for DevX?